internal/services/workloads/workloads_sap_virtual_instance_resource_test.go

```diff
@@ -0,0 +1,1122 @@
+		return nil, err
+	}
+
+	client := clients.Workloads.SAPVirtualInstances
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r WorkloadsSAPVirtualInstanceResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+data "azurerm_subscription" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-sapvis-%d"
+  location = "%s"
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "acctest-uai-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_role_assignment" "test" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Azure Center for SAP solutions service role"
+  principal_id         = azurerm_user_assigned_identity.test.principal_id
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-vnet-%d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctest-subnet-%d"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_resource_group" "app" {
+  name     = "acctestRG-sapapp-%d"
+  location = "%s"
+
+  depends_on = [
+    azurerm_subnet.test
+  ]
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r WorkloadsSAPVirtualInstanceResource) basic(data acceptance.TestData, sapVISNameSuffix int) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_workloads_sap_virtual_instance" "test" {
+  name                        = "X%d"
+  resource_group_name         = azurerm_resource_group.test.name
+  location                    = azurerm_resource_group.test.location
+  environment                 = "NonProd"
+  sap_product                 = "S4HANA"
+  managed_resource_group_name = "managedTestRG%d"
+
+  deployment_with_os_configuration {
+    app_location = azurerm_resource_group.app.location
+
+    os_sap_configuration {
+      sap_fqdn = "sap.bpaas.com"
+    }
+
+    single_server_configuration {
+      app_resource_group_name = azurerm_resource_group.app.name
+      subnet_id               = azurerm_subnet.test.id
+      database_type           = "HANA"
+      secondary_ip_enabled    = true
+
+      virtual_machine_configuration {
+        vm_size = "Standard_E32ds_v4"
```

Comment:
this is *huge* (and thus, incredibly expensive to run) - is this the minimum we can get away with here?

Reply:
Yes. I have set the minimum values in acctest.